### PR TITLE
Fix parentheses around infix operators with attributes

### PR DESCRIPTION
**Pull request by CraigFe**
This resolves https://github.com/ocaml-ppx/ocamlformat/issues/1250.

As a side-effect, there are now unnecessary parentheses around infix application chains with attributes attached, e.g.:

```diff
-;; a + b + c [@attr]
+;; (a + b + c) [@attr]
```

This is a breaking change. I chose to propose the change anyway, since I find the precedence binding of attributes to infix application confusing. For instance:

```ocaml
(* Here the attribute binds to [(a::b)] *)
let _ = (a :: b [@attr])

(* Here the attribute binds to [b] only *)
let _ = (a @ b [@attr])
```

If we decide to keep only the parentheses necessary, I think it's necessary to add attributes to the infix precedence logic. In that case, we might want to consider what the interaction with the `infix-precedence` option should be.

Let me know what you think.

### CODE REVIEW

**Comment by gpetiot** [comment]
I removed the unnecessary parentheses, no regression with test_branch. I added an ugly test of the context that should be merged with the parenze functions in Ast but it's such a mess right now that I didn't manage to do it.

File: test/passing/attributes.ml [hunk 1]
```diff
@@ -209,30 +209,29 @@ let _ = (module M : S) [@a]
...
 let _ = f ((module M : S) [@a])
 
-let _ = (([] @ []) [@a])
+let _ = ([] @ []) [@a]
 
 (* Infix operator should left-align with the inner parens *)
 let _ =
-  (( a_____________________________________
-   @ b_____________________________________ ) [@a])
+  ( a_____________________________________
```
Comment (CraigFe) [inline test/passing/attributes.ml]: This test now must be changed to remain useful.

Comment (CraigFe) [reply test/passing/attributes.ml]: Done.

**Comment by Julow** [comment]
Thanks for the work !

**Pull request merged**